### ISSUE TITLE
I wish it would use controlnet! I love it so much!

### ISSUE THREAD

**Issue by shiyk92** [lllyasviel/Fooocus#46]
Are there any plans for expansion in the near future?

**Comment by Tigwin** [comment]
yeah, I'd love for it to have controlnet too, but at some point they'd be recreating a1111 and we lose the simple interface. I like them going for a new market - a MJ style replacement, with similar options and features.

**Comment by reese555** [comment]
i think he means to keep it streamlined, not recreate a111,sdnet or invoke. to me this is a stepping stone for those who would be scared away by denoise,samplers etc. it just works and its absolutely perfect for my son without risk of him breaking everything.

**Comment by lllyasviel** [comment]
We will only officially maintain ControlNet in A1111 environment and pure python.

Project Fooocus will not have ControlNet in it.

Nevertheless, we will pick several most useful features from midjourney to implement, and make sure that their quality is the best, comparing to all open-source projects. 

Typically, the next priority is midjourney’s “variation”, “scale”, and “image prompt”. And even for those features, adding any extra things will be extremely careful.

We promise that the main UI will never change. What you get in the first place is always one textbox and one button.

**Comment by boombbo** [comment]
你的逻辑美学中，缺少了alpha：^ _ ^> We will only officially maintain ControlNet in A1111 environment and pure python.
> 
> 
> 
> Project Fooocus will not have ControlNet in it.
> 
> 
> 
> Nevertheless, we will pick several most useful features from midjourney to implement, and make sure that their quality is the best, comparing to all open-source projects. 
> 
> 
> 
> Typically, the next priority is midjourney’s “variation”, “scale”, and “image prompt”. And even for those features, adding any extra things will be extremely careful.
> 
> 
> 
> We promise that the main UI will never change. What you get in the first place is always one textbox and one button.
> 
>

**Comment by adafym** [comment]
Well, maybe you can try[ Fooocus-ControlNet-SDXL](https://github.com/fenneishi/Fooocus-ControlNet-SDXL), I found this in [Fooocus Advanced #117](https://github.com/lllyasviel/Fooocus/discussions/117) and tested it, it works great.

**Comment by tomasklaen** [comment]
> Project Fooocus will not have ControlNet in it.

I have to say this is incredibly disappointing.

And I'm not alone in this. Most of the mentions of Foocus I've seen thus far, [including youtube tutorials](https://youtu.be/w5EA358IPCs?t=346), everyone notes something along the lines of "they don't seem to have controlnets yet which is unfortunate".

I don't see how it would be any more complicated than existing advanced features already are. Just a drag and drop of an image and a couple options next to it. Is the main argument against it that midjourney doesn't have this feature, so you're going to cripple your tool in the same way?

Or do you think this is your average user, and they can't handle the concept of controlling the outputs with controlnets?

![average_user](https://github.com/lllyasviel/Fooocus/assets/47283320/5b20a116-1300-4251-97b7-653c5ec79db5)

Again, incredibly disappointing.

_I know about the Fooocus-Control fork, but using a single guy's fork of a main project is rarely a good idea in the long run._
